pydantic/validators.py

```diff
@@ -244,4 +252,7 @@ def find_validators(type_):
                 return validators
         except TypeError as e:
             raise RuntimeError(f'error checking inheritance of {type_!r} (type: {display_as_type(type_)})') from e
-    raise errors.ConfigError(f'no validator found for {type_}')
+    if arbitrary_types_allowed:
+        return [make_arbitrary_type_validator(type_)]
+    else:
+        raise errors.ConfigError(f'no validator found for {type_}')
```

Comment:
This should be a RuntimeError, it's not a config problem and will happen at runtime.